.github/workflows/unittests.yml

```diff
@@ -0,0 +1,83 @@
+name: Unittests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  actions: write  # Needed for skip-duplicate-jobs job
+  contents: read
+  checks: write   # Needed for publish test results job
+  pull-requests: write   # Needed for publish test results job
+
+jobs:
+  # Special job which automatically cancels old runs for the same branch, prevents runs for the
+  # same file set which has already passed, etc.
+  pre_job:
+    name: Skip Duplicate Jobs Pre Job
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@9d116fa7e55f295019cfab7e3ab72b478bcf7fdd # v4.0.0
+        with:
+          cancel_others: 'true'
+          github_token: ${{ github.token }}
+
+  unittests:
+    name: Unittests - Python ${{ matrix.python-version }} - ${{ matrix.platform }}
+    runs-on: ${{ matrix.platform }}-latest
+    needs: pre_job
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.5"
+          - "3.6"
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+        platform:
+          - ubuntu
+          - macos
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install tox
+        env:
+          TOX_VERSION: 3.20.1
+        run: |
+          python -m pip install --upgrade pip
+          pip install "tox==$TOX_VERSION" pytest-github-actions-annotate-failures
+
+      - name: Run Unit Tests
+        run: tox -e "py${{ matrix.python-version }}-unit-tests"
+
+      - name: Upload pytest test results
+        uses: actions/upload-artifact@v3
+        with:
+          name: pytest-results-${{ matrix.python-version }}
+          path: test-results/junit*.xml
+        # Use always() to always run this step to publish test results when there are test failures
+        if: ${{ always() }}
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        if: ${{ always() }}
+        with:
+          junit_files: "test-results/junit*.xml"
+          comment_title: Test Results - Python ${{ matrix.python-version }} - ${{ matrix.platform }}
```

Comment:
I think having comment per job is a bit too spammy - is there a way to consolidate results of all the jobs in a single comment?

Reply:
Yes: https://github.com/scalyr/scalyr-agent-2/pull/950#issuecomment-1208734836
I don't know if it will properly handle linking back to the original test properly in this configuration. I guess we'll just have to wait and see....